examples/build_image.sh

```diff
@@ -14,25 +14,94 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# This env variables can help when behind corporate network
+# export GOINSECURE="dmitri.shuralyov.com"
+# export GOPRIVATE=*
+
+helpFunction()
+{
+   echo ""
+   echo "Usage: $0 -t type"
+   echo -e "\t-t The type parameters allows to select which Sedna example to build (joint_inference, federated_learning, etc..)"
+   exit 1 # Exit script after printing help
+}
+
+while getopts "t:" opt
+do
+   case "$opt" in
+      t ) type="$OPTARG" ;;
+      ? ) helpFunction ;; # Print helpFunction in case parameter is non-existent
+   esac
+done
+
+if [ -z "$type" ]
+then
+   echo "Defaulting to building all example images..";
+   type="all"
+fi
+
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
 IMAGE_REPO=${IMAGE_REPO:-kubeedge}
 IMAGE_TAG=${IMAGE_TAG:-v0.4.0}
-
 EXAMPLE_REPO_PREFIX=${IMAGE_REPO}/sedna-example-
 
-dockerfiles=(
+# Uncomment this line if you want to push your images to a private repository
+# PRIVATE_DOCKER_REPOSITORY="my-repo.com"
+
+dockerfiles_federated_learning=(
 federated-learning-mistnet-yolo-aggregator.Dockerfile
 federated-learning-mistnet-yolo-client.Dockerfile
 federated-learning-surface-defect-detection-aggregation.Dockerfile
-federated-learning-surface-defect-detection-train.Dockerfile
-incremental-learning-helmet-detection.Dockerfile
+federated-learning-surface-defect-detection-aggregation-train.Dockerfile
+)
+
+dockerfiles_joint_inference=(
 joint-inference-helmet-detection-big.Dockerfile
 joint-inference-helmet-detection-little.Dockerfile
+)
+
+dockerfiles_lifelong_learning=(
 lifelong-learning-atcii-classifier.Dockerfile
 )
 
+dockerfiles_incremental_learning=(
```

Comment:
Can you improve it like this?
By providing subdirectories, this script detects these subdirectories' dockerfile and build it

Reply:
no subdirectories provding means all subdirectories

Reply:
The developer can be very easy to provide subdirectories just by click `Tab` keystroke in interactive terminal.